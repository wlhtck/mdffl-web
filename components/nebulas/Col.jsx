import styled from 'styled-components';
import { oneOf, bool, number } from 'prop-types';

const Col = styled.div`
  align-items: ${({ align }) => (align)};
  display: flex;
  flex-direction: ${({ reverse }) => (reverse ? 'column-reverse' : 'column')};
  justify-content: center;
  width: ${({ padding }) => (padding ? `calc(50% - ${padding * 2}px)` : '50%')};
  ${({ padding }) => (padding ? `padding: ${padding}px;` : '')}

  *:last-child {
    margin-bottom: 0;
  }
`;

Col.propTypes = {
  align: oneOf(['center', 'flex-start', 'flex-end']),
  reverse: bool,
  padding: number,
};

Col.defaultProps = {
  align: 'center',
  reverse: false,
  padding: null,
};

export default Col;
