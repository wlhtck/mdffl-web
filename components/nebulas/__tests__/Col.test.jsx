import React from 'react';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import Col from '../Col';

describe('components/nebulas/Col', () => {
  it('renders a default <Col />', () => {
    expect(shallow(<Col />)).toMatchSnapshot();
  });

  it('renders a flex-start <Col />', () => {
    expect(shallow(<Col align="flex-start" />)).toMatchSnapshot();
  });

  it('renders a flex-end <Col />', () => {
    expect(shallow(<Col align="flex-end" />)).toMatchSnapshot();
  });

  it('renders a reverse <Col />', () => {
    expect(shallow(<Col reverse />)).toMatchSnapshot();
  });

  it('renders a padded <Col />', () => {
    expect(shallow(<Col padding={100} />)).toMatchSnapshot();
  });
});
