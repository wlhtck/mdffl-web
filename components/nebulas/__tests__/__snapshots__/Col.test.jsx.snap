// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/nebulas/Col renders a default <Col /> 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 50%;
}

.c0 *:last-child {
  margin-bottom: 0;
}

<div
  className="c0"
/>
`;

exports[`components/nebulas/Col renders a flex-end <Col /> 1`] = `
.c0 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 50%;
}

.c0 *:last-child {
  margin-bottom: 0;
}

<div
  className="c0"
/>
`;

exports[`components/nebulas/Col renders a flex-start <Col /> 1`] = `
.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 50%;
}

.c0 *:last-child {
  margin-bottom: 0;
}

<div
  className="c0"
/>
`;

exports[`components/nebulas/Col renders a padded <Col /> 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 50%;
  padding: 100px;
}

.c0 *:last-child {
  margin-bottom: 0;
}

<div
  className="c0"
/>
`;

exports[`components/nebulas/Col renders a reverse <Col /> 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 50%;
}

.c0 *:last-child {
  margin-bottom: 0;
}

<div
  className="c0"
/>
`;
